### PR TITLE
Update Workflow with CodeQL

### PR DESCRIPTION
**Pull request by hillmich**
- Add workflow for CodeQL (superseeds LGTM)
- Add workflow concurrency settings

### CODE REVIEW

**Review by burgholzer — APPROVED** [review]
Would be nice to also fix or dismiss the newly brought up alerts. Otherwise this LGTM 👍🏼

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -2,41 +2,54 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [ master ]
+    branches: [ main ]
   pull_request:
-    branches: [ master ]
-  schedule:
-    - cron: '15 21 * * 6'
```
Comment (burgholzer) [inline .github/workflows/codeql-analysis.yml:9]: Did you remove the scheduled run on purpose?

**Review by burgholzer — APPROVED** [review]
LGTM 👍🏻 one minor question. Feel free to resolve and merge.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (hillmich) [reply .github/workflows/codeql-analysis.yml:9]: Yes, it's intentional. I don't see much value in running the check every Saturday. Do the queries and checks of CodeQL get updated that regularly?

Comment (burgholzer) [reply .github/workflows/codeql-analysis.yml:9]: I guess it's not that critical for us, but it gets regularly updated with security issues and potential fixes.

Comment (hillmich) [reply .github/workflows/codeql-analysis.yml:9]: I share the opinion that is it's not (time-) critical for us. In the spirit of having dependencies only updated automatically once per month, I think running the checks only on PR and on main is sufficient.

**Pull request merged**